File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        logLine = logLine.Trim();
        int messageStart = logLine.IndexOf(" ") + 1;
        return logLine.Substring(messageStart).Trim();
    }

    public static string LogLevel(string logLine)
    {
        logLine = logLine.Trim();
        return logLine.Substring(1, logLine.IndexOf("]") - 1).Trim().ToLower();
    }

    public static string Reformat(string logLine)
    {
        return Message(logLine) + " (" + LogLevel(logLine).ToLower() + ")";
    }
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        // sutract level and template characters
        String[] split = logLine.split(":");

        // strip message of left and right white spaces
        String message = split[1].trim();

        // return message
        return message;
    }

    public static String logLevel(String logLine) {
        String[] split = logLine.split(":");

        String logLevel = split[0].replace("[", "").replace("]", "");

        return logLevel.toLowerCase();
    }

    public static String reformat(String logLine) {
        return "%s (%s)".formatted(message(logLine), logLevel(logLine));
    }
}
